### DISCUSSION TITLE
thanks

### DISCUSSION THREAD

**Discussion by 73rrto4r** [tp86/legimi#15] [General]
Thanks for the work.

**Comment by tp86** [comment]
Thank you for your feedback, I really appreciate it.

I see in the email notification that you had some questions regarding deviceid, have you figured it out yourself? Can I help you with something?

BTW, I'm currently working on second version of script, written in Go. It should be easier to install and use and hopefully have fewer bugs. It's not finished yet, but work is already fairly advanced, I think first usable version will be available in a couple of weeks.

**Reply by 73rrto4r** [reply]
Hi,
Thanks for the work, your code is neat.
I didn't manage to make the script to work, but I thought it will be just waste of your time. I believe that osx causes issues especially somewhere with libraries. Probably some small tweak somwhere, a bad path etc. If you can tell me what is in the deviceid I could give it another go (with low chances of success) - at the same time I am happy to wait for your new version 2.0 :-)

**Reply by tp86** [reply]
In case of OSX, you'll probably be better off with the official Legimi app. I designed the script for use on Linux which does not have official app.

That said, for OSX there should not be much difference in usage of Lua, but I'm not the expert and don't have the system to test things out.

For device id, it is the internal id that Legimi associates with your Kindle, I don't know if there is other way to obtain it than querying the API. Once you have it, it should be stored in file called `deviceid` in the same directory as script.

Finally, version 2.0 will be self-contained Go binary, so it should work the same on all supported platforms.

**Reply by 73rrto4r** [reply]
I was hoping for quick switching option between accounts, as at the moment I have to have a separate accounts on the Mac for each family member, (or I could log out from the app to synch their books). I came across your app, and I thought it might be easier, plus why not to try something else.
I was considering Linux as a main os / new pc, and investing Legimi support on Linux resulted in your code. Great job, otherwise there would be no Legimi on Linux. I'm curious what will you make with 2.0. Good luck!

**Reply by tp86** [reply]
> I'm curious what will you make with 2.0.

Basic version will have the same functionality as Lua script: listing books on your shelf and downloading them. I'm not planning to create clone of official Legimi app. You will still have to copy downloaded file to Kindle yourself, for example.

However, I'm planning to add switch for giving different locations of configuration file/directory which will store device id and optionally credentials (not secured), so your use case will probably be possible.
